src/Model.jl

```diff
@@ -85,8 +88,7 @@ function Model(lattice::AbstractMatrix{T};
                spin_polarization=default_spin_polarization(magnetic_moments),
                symmetries=default_symmetries(lattice, atoms, magnetic_moments, terms, spin_polarization),
                ) where {T <: Real}
-    lattice = Mat3{T}(lattice)
```

Comment:
This line is still needed, actually to do the conversion to the type we use internally in DFTK. That's why some of the tests  fail.